src/Buffer.cpp

```diff
@@ -3,13 +3,19 @@
 #include "Error.h"
 #include "JITModule.h"
 #include "runtime/HalideRuntime.h"
+#include "Target.h"
 
 namespace Halide {
 namespace Internal {
 
 namespace {
 void check_buffer_size(uint64_t bytes, const std::string &name) {
-    user_assert(bytes < (1UL << 31)) << "Total size of buffer " << name << " exceeds 2^31 - 1\n";
+    Target t = get_target_from_environment();
```

Comment:
This is about being able to construct an actual buffer in memory right now. I think the check should be if sizeof(void *) == 8. get_target_from_environment can easily return the wrong thing if you're cross-compiling 64-bit code from a 32-bit environment.